### PR TITLE
PHP8: add crashing nullable type case

### PR DESCRIPTION
**Pull request by keradus**
anyone willing to take over?

### CODE REVIEW

File: tests/Fixtures/Integration/misc/property_promotion_with_nullable_type.test [hunk 1]
```diff
@@ -0,0 +1,20 @@
+--TEST--
+Integration test to verify that property promotion with nullable type does not crash the Tokenizer.
+--RULESET--
+{
+    "ternary_to_elvis_operator": true
```
Comment (keradus) [inline tests/Fixtures/Integration/misc/property_promotion_with_nullable_type.test:5]: random rule. not related to the problem...
most probably the test case can be moved directly to `TokenizerTest` class

**Comment by Wirone** [comment]
Most probably related to #5404 :slightly_smiling_face:

**Comment by keradus** [comment]
lets restart the CI and confirm that assumption. but looks promising indeed

**Comment by Wirone** [comment]
In general, after #5404 tokenizer works properly with property promotion and nullable types and as long as test cases have proper `@require` with PHP version, it will work as expected. It will fail like `Parse error: syntax error, unexpected 'private' (T_PRIVATE), expecting variable (T_VARIABLE)` in runtime <8.0, but this fail is from `AbstractFixerTestCase::lintSource()` just before tokenization.

Side questions:
- Why `AbstractFixerTestCase::lintSource()` catches `\Exception`, not `\Throwable`?
- What about [failed tests](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5400/checks?check_run_id=1628196122#step:10:450), why check is marked as passed :thinking: (out of curiosity)?

**Comment by keradus** [comment]
> It will fail like Parse error: syntax error, unexpected 'private' ...

@Wirone , can you please point to cases where `@requires PHP 8.0` is missing? we should add it for code-samples relying on PHP 8 syntax.

> Why `AbstractFixerTestCase::lintSource()` catches `\Exception`, not `\Throwable`?

probably code create before `Throwable` was introduced. Do we have a valid case when `Throwable` is raised and we should catch it, rather than fix code leading to this exception?

> - What about [failed tests](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5400/checks?check_run_id=1628196122#step:10:450), why check is marked as passed :thinking: (out of curiosity)?

I did rebase and cannot see the link content anymore. Anyway, could be related to GitHub Action configuration: `continue-on-error: ${{ matrix.php-version == '8.0' }}`

**Comment by keradus** [comment]
works smoothly now, looks like #5404 saved the day indeed!

> OK, but incomplete, skipped, or risky tests!
> Tests: 24609, Assertions: 540397, Skipped: 180, Incomplete: 26.

**Comment by Wirone** [comment]
Sorry for radio silence, I missed this notification. I did not see any `@requires PHP 8.0` missing, I just said that it will fail IF run on the wrong version of PHP. Good to hear that it's working :slightly_smiling_face:

**Pull request closed without merging**